### ISSUE TITLE
ENH: nice logging / printing for multiple channel types.

### ISSUE THREAD

**Issue by dengemann** [autoreject/autoreject#68]
Sequel to #64

**Comment by jasmainak** [comment]
What do you mean there is no way way to control verbosity? There is a `verbose` parameter, no?

**Comment by larsoner** [comment]
Not here:

https://github.com/autoreject/autoreject/blob/master/autoreject/autoreject.py#L140

And it uses `print`

**Comment by jasmainak** [comment]
okay, we could fix this. 

I remember running into issues with the verbosity decorator in MNE because it didn't support progressbars ...

so would prefer a simple if statement based verbose

**Comment by larsoner** [comment]
IIRC in MNE we use `print` only for progressbars (because they delete) and `verbose` for everything else because it routes through `logger`, which gives lots of advantages (e.g., standard file-output setting and the like)

**Comment by jasmainak** [comment]
Yeah I saw that. But some people wanted both: i.e., progressbars but also the ability to suppress them. At the moment, `autoreject` is not so fast that we could do away with the progressbars ...

**Comment by larsoner** [comment]
We also do something like this in MNE. With the `verbose` decorator, you just see if `logger.level <= logging.INFO` and if so call your `print(...)`, if not, don't. Then `verbose=False` or `verbose='warning'` won't do progressbar and `verbose=True` or `verbose='info'` will. We do this in our dataset fetchers.

**Comment by jasmainak** [comment]
hmm ... that might be a solution indeed
